src/features/element-hiding.js

```diff
@@ -25,13 +27,12 @@ function collapseDomNode (element, rule, previousElement) {
     }
     const type = rule.type
     const alreadyHidden = hiddenElements.has(element)
-
-    if (alreadyHidden) {
+    const alreadyModified = modifiedElements.has(element) && modifiedElements.get(element) === rule.type
+    // return if the element has already been hidden, or modified by the same rule type
+    if (alreadyHidden || alreadyModified) {
         return
     }
 
-    featureInstance.addDebugFlag()
```

Comment:
This was firing every time `collapseDomNode` runs, so I moved it into `hideNode` so that it only fires when something is actually hidden; and I added a check so it only fires once.